java-frontend/src/main/java/org/sonar/java/regex/ast/RegexBaseVisitor.java

```diff
@@ -52,10 +47,38 @@ protected JavaCharacter getJavaCharacterForFlag(int flag) {
     return activeFlags.getJavaCharacterForFlag(flag);
   }
 
+  private void visit(RegexTree tree) {
+    tree.accept(this);
+  }
+
   private void visit(List<RegexTree> trees) {
     trees.forEach(this::visit);
   }
 
+  @Override
+  public void visit(RegexParseResult regexParseResult) {
+    if (!regexParseResult.hasSyntaxErrors()) {
+      activeFlags = regexParseResult.getInitialFlags();
+      before(regexParseResult);
```

Comment:
This one is actually not used. Do you need it for a coming rule?
If it's not the case I would simply drop it for the time being (less is more).